File: backend/src/main/java/backend/resource/bean/ListParam.java
package backend.resource.bean;

import javax.ws.rs.DefaultValue;
import javax.ws.rs.QueryParam;

import lombok.Data;

/** Data Transfer Object for request parameters for list resources. */
@Data public class ListParam {
	@DefaultValue("3")
	@QueryParam("size")
	private Integer size;

	@DefaultValue("0")
	@QueryParam("page")
	private Integer page;
}

File: backend/src/main/kotlin/backend/resource/bean/ListParam.kt
package backend.resource.bean

import javax.ws.rs.DefaultValue
import javax.ws.rs.QueryParam

/** Data Transfer Object for request parameters for list resources.  */
data class ListParam (
    @DefaultValue("10")
    @QueryParam("size")
    val size: Int = 0,

    @DefaultValue("0")
    @QueryParam("page")
    val page: Int = 0
)